### PR TITLE
[Account] Added new test for rotate auth key 

### PR DESCRIPTION
**Pull request by chloeqjz**
### Description
- updated rotate auth key tests
previously, I was using `harness.new_account_at(AccountAddress::from_hex_literal(address).unwrap())` to create new accounts in the tests, but all accounts created this way are created with the same key pair, so the test was not exactly accurate. 
also added test to rotate auth key twice - previously, after rotating from `auth_key_a` to `auth_key_b`, it was not clear if we should call `rotate_authentication_key_ed25519` from `account_a` or `account_b`. the new test clarifies that we intend to call `rotate_authentication_key_ed25519` from `account_a` (the originating account). 

- updated account.move
previously, after rotating from `auth_key_a` to `auth_key_b`, we can get the `originating_address` if we call `rotate_authentication_key_ed25519` from `account_a`. I changed the code so that we can get the `originating_address` from either `account_a` or `account_b` (so now it's up to the developers which account to call this function from).

### Test Plan
unit tests, e2e tests

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/3483)

### CODE REVIEW

File: aptos-move/e2e-move-tests/tests/rotate_auth_key.rs [hunk 1]
```diff
@@ -88,3 +88,107 @@ fn rotate_auth_key() {
...
+        new_public_key: new_public_key2.to_bytes().to_vec(),
+    };
+
+    let rotation_msg2  = bcs::to_bytes(&rotation_proof2);
+
+    let rotation_proof_signed_by_current_private_key2 = account1.privkey.sign_arbitrary_message(&rotation_msg2.clone().unwrap());
+    let rotation_proof_signed_by_new_private_key2 = new_private_key2.sign_arbitrary_message(&rotation_msg2.unwrap());
+
+    assert_success!(harness.run_transaction_payload(
+        &account1,
```
Comment (areshand) [inline aptos-move/e2e-move-tests/tests/rotate_auth_key.rs:171]: Please try rotate with &account2 and see what happens. 
The scenario is "the owner of account1 rotates to account2's auth_key" 
Now, owner use account2's key and use it for further rotation

File: aptos-move/e2e-move-tests/tests/rotate_auth_key.rs [hunk 2]
```diff
@@ -31,48 +31,82 @@ struct RotationProofChallenge {
...
+    assert_successful_payload_key_rotation(
+        &mut harness,
+        account1.clone(),
+        *account1.address(),
+        10,
+        account2.privkey.clone(),
+        account2.pubkey.clone(),
+    );
+    // rotate account1's keypair to account2
+    account1.rotate_key(account2.privkey, account2.pubkey);
```
Comment (areshand) [inline aptos-move/e2e-move-tests/tests/rotate_auth_key.rs:49]: @jjleng do we plan to support updating the original account's private key and public key after key rotation in cmd and SDK?

**Review by areshand — APPROVED** [review]

File: aptos-move/e2e-move-tests/tests/rotate_auth_key.rs (hunk 2 above)
Comment (jjleng) [reply aptos-move/e2e-move-tests/tests/rotate_auth_key.rs:49]: for CLI, users can save the new pri_k and pub_k into a new profile. We don't override the existing profile.

File: aptos-move/framework/aptos-framework/sources/account.move [hunk 3]
```diff
@@ -239,8 +239,9 @@ module aptos_framework::account {
         // Update the originating address map: i.e., set this account's new address to point to the originating address.
         // Begin by removing the entry for the current authentication key, if there is one.
         let address_map = &mut borrow_global_mut<OriginatingAddress>(@aptos_framework).address_map;
+        let originating_address = addr;
         if (table::contains(address_map, curr_auth_key)) {
-            table::remove(address_map, curr_auth_key);
+            originating_address = table::remove(address_map, curr_auth_key);
```
Comment (jjleng) [inline aptos-move/framework/aptos-framework/sources/account.move:244]: I believe this is the correct behavior

**Review by jjleng — APPROVED** [review]

**Pull request merged**